src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -90,6 +90,7 @@ private void EnsureApplicationServices()
         {
             EnsureStartup();
 
+            _applicationServiceCollection.AddSingleton<IHostingKeepAlive, ConsoleHostingKeepAlive>();
```

Comment:
This implementation should be registered in Program.cs. Helios (the IIS layer) also uses this library.